File: src/strategy/javascript/check.js
exports.Check = function () {
    var _items = [];
    
    this.pushItem = function(item){
        _items.push(item);
    }

    this.getItems = function()
    {
        return _items; 
    }
    this.getTotal = function(){
        var total = 0.0;
        for (item of _items){
            total+=item.getValue();
        }
        return total;    
    }
}

exports.Item = function(name, value){
    var _name = name;
    var _value = value;

    this.getName = function(){
        return _name;
    }

    this.getValue = function(){
        return _value;
    }
}

File: src/strategy/c#/Check.cs
using System;
using System.Collections.Generic;

namespace Strategy {
    class Check{
        public List<Item> items {get;}

        public Check(){
            items = new List<Item>();
        }

        public void pushItem(Item item){
            items.Add(item);
        }

        public double getTotal(){
            double total = 0.0;
            foreach (Item item in items)
            {
                total+=item.value;
            }
            return total;
        }
    }


    class Item
    {
        public Item(string name_, double value_){
            name = name_;
            value = value_;
        }
        public string name {get;set;}
        public double value {get;set;}

    }
}